### ISSUE TITLE
KeyError: 'AllowedIPs'

### ISSUE THREAD

**Issue by tonjo** [donaldzou/wireguard-dashboard#9]
UI starts, then when I click on `wg0`, blank page and the error.
_(Note: wireguard is perfectly working)_

Full traceback:
```
Traceback (most recent call last):
  File "/opt/wgdashboard/lib/python3.7/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/opt/wgdashboard/lib/python3.7/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/opt/wgdashboard/lib/python3.7/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/opt/wgdashboard/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/opt/wgdashboard/lib/python3.7/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/opt/wgdashboard/lib/python3.7/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "dashboard.py", line 286, in get_conf
    "peer_data": getdb(config_name),
  File "dashboard.py", line 172, in getdb
    get_conf_peers_data(config_name)
  File "dashboard.py", line 169, in get_conf_peers_data
    db.update({"allowed_ip":i['AllowedIPs']}, peers.id == i["PublicKey"])
KeyError: 'AllowedIPs'
```

**Comment by donaldzou** [comment]
Could you provide a sample of your wg configuration file please?

**Comment by tonjo** [comment]
Below a sample

```
[Peer]
PublicKey = XXXXXXXXXXXXXXXXXXXXXXXXX
AllowedIPs = 10.2.1.3/32
Endpoint = IP:PORT

[Peer]
PublicKey = XXXXXXXXXXXXXXXXXXXXXXXXX
AllowedIPs = 10.2.1.4/32

[Peer]
PublicKey = XXXXXXXXXXXXXXXXXXXXXXXXX
AllowedIPs = 10.2.1.5/32
Endpoint = IP:PORT
```

**Comment by donaldzou** [comment]
Based on the error, it is not getting the `AllowedIPs` correctly from the `.conf` file, I'll look into it.

**Comment by donaldzou** [comment]
I have a theory, i think it may not have the permission to open your `.conf` file, are you using `root` to run the python file? I'm also wondering if the error you posted is all the error is showing?

Thank you!

**Comment by tonjo** [comment]
I am running as root, That is the only error I got, I just skipped a couple of lines before the error.
Below the complete output.
Note: I am seeing this dashboard through the VPN itself.

```
# python dashboard.py 

 * Serving Flask app "Wireguard Dashboard" (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://0.0.0.0:10086/ (Press CTRL+C to quit)
[2021-04-09 10:37:44,872] ERROR in app: Exception on /get_config/wg0 [GET]
Traceback (most recent call last):
  File "/opt/wgdashboard/lib/python3.7/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/opt/wgdashboard/lib/python3.7/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/opt/wgdashboard/lib/python3.7/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/opt/wgdashboard/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/opt/wgdashboard/lib/python3.7/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/opt/wgdashboard/lib/python3.7/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "dashboard.py", line 286, in get_conf
    "peer_data": getdb(config_name),
  File "dashboard.py", line 172, in getdb
    get_conf_peers_data(config_name)
  File "dashboard.py", line 169, in get_conf_peers_data
    db.update({"allowed_ip":i['AllowedIPs']}, peers.id == i["PublicKey"])
KeyError: 'AllowedIPs'
10.2.1.5 - - [09/Apr/2021 10:37:44] "GET /get_config/wg0 HTTP/1.1" 500 -

```

**Comment by tonjo** [comment]
My sample conf was not complete, I did not see a Peer without AllowedIps. It can be a misconfiguration, but giving the fact wireguard don't complain, it should not broke, so I proposed a working fix (I can see all peers now).

**Comment by donaldzou** [comment]
I think your #11 would solve this problem in the future, so you didn't sent endpoint in your `wg0.conf` for all of your peers and it still works right?

**Comment by tonjo** [comment]
Yes, there is on the server side one single endpoint without AllowedIps, **that particular peer probably will never work**, 
but all the others works perfectly fine. I noticed that misconfiguration only when I tried your nice dashboard.
So my point is: it could be a misconfiguration, but wireguard still works, the dashboard does not. Why?

A good idea could be: showing a warning "NOTE: Peer wihtout AllowedIps".

A more correct sample of my configuration:
```
[Interface]
Address = 10.2.0.1/32
SaveConfig = true
PostUp = ip route add 10.2.0.0/16 dev wg0 scope link
PostDown = ip route del 10.2.0.0/16 dev wg0 scope link
ListenPort = 1194
PrivateKey = XXXXXXXXXXXXXXXXXXXXXXXXX

[Peer]
PublicKey = XXXXXXXXXXXXXXXXXXXXXXXXX
AllowedIPs = 10.2.1.3/32
Endpoint = IP:PORT

[Peer]
PublicKey = XXXXXXXXXXXXXXXXXXXXXXXXX
AllowedIPs = 10.2.1.4/32

[Peer]
PublicKey = XXXXXXXXXXXXXXXXXXXXXXXXX          # <--- Note, peer with no AllowedIps

[Peer]
PublicKey = XXXXXXXXXXXXXXXXXXXXXXXXX
AllowedIPs = 10.2.1.5/32
Endpoint = IP:PORT
```

**Comment by donaldzou** [comment]
Ohh I get it now, yeah WG works fine even with peer without `AllowedIPs`, thank you for providing this information!

**Comment by aakashjonwal89** [comment]
Hi 

We have solve this issues after adding the parameter "SaveConfig = true" in wg0.conf.
Wiregaurd-Dashboard is not able to save the config into its own db due to config not persisted. So you need to add this parameter in the wg0.conf file. 

Sample:
[Interface]
Address = 10.0.12.130/32
Address = fe80::f743:8b97:abea:4ff2/64  ------------- this address added by the wireguard dashboard
SaveConfig = true
ListenPort = 51820
PrivateKey = XXXXXXXXXXXXXXXXXXXXXX

[Peer]
PublicKey = XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
AllowedIPs = 10.0.12.189/32

![image](https://user-images.githubusercontent.com/12555644/119841415-6689ce80-bf23-11eb-99dd-ecc53d23ca07.png)

Note: Whenever you will start wg-quick and dashboard, Please remove the file from "Wireguard-Dashboard/src/db", using below command:
$ rm -rf Wireguard-Dashboard/src/db*

**Comment by donaldzou** [comment]
Hi @aakashjonwal89 , i think i know what happened. In the first message you sent me, in your sample config file, you have `Publickey` but it should be `PublicKey`, which is the the way that wireguard and wireguard-dashboard read the config of each peer and i think wireguard itself fixed this problem and wireguard-dashboard run at no problem now :)

**Comment by robchez** [comment]
This resolved my problem, see Issue#21 was never able to get it to work. Added  **SaveConfig = true** and deleted db Restarted WG and WG Dashboard, working like a champ!!

`$ rm -rf Wireguard-Dashboard/src/db/*` in [comment](https://github.com/donaldzou/wireguard-dashboard/issues/9#issuecomment-849669447) above.
